### PR TITLE
Move to TypeScript and simplify GlimmerApp

### PR DESCRIPTION
**Pull request by tomdale**
* Removes some inherited configuration stuff I’m not sure if we need
* Build pipeline is a little simpler and hopefully faster now
* Drops ember-cli dependency

### CODE REVIEW

File: lib/broccoli/glimmer-app.js [hunk 1]
```diff
@@ -1,293 +0,0 @@
...
-};
-
-module.exports = GlimmerApp;
-
-/**
- * GlimmerApp provides an interface to a package (app, engine, or addon)
- * compatible with the module unification layout.
- *
- * @class GlimmerApp
- * @extends EmberApp
```
Comment (dgeb) [inline lib/broccoli/glimmer-app.js]: GlimmerApp no longer extends EmberApp

File: lib/broccoli/glimmer-app.ts [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+interface Tree {
+
+}
+
+/**
+ * GlimmerApp provides an interface to a package (app, engine, or addon)
+ * compatible with the module unification layout.
+ *
+ * @class GlimmerApp
+ * @extends EmberApp
```
Comment (dgeb) [inline lib/broccoli/glimmer-app.ts]: GlimmerApp no longer extends EmberApp

File: lib/broccoli/glimmer-app.ts [hunk 3]
```diff
@@ -0,0 +1,358 @@
...
+      compress: {
+        screw_ie8: true,
+      },
+      sourceMapConfig: {
+        enabled: false
+      }
+    });
+  }
+
+  _configPath() {
```
Comment (dgeb) [inline lib/broccoli/glimmer-app.ts]: private or protected

File: lib/broccoli/glimmer-app.ts [hunk 4]
```diff
@@ -0,0 +1,358 @@
...
+      configPath: path.join(this.name, 'config', 'environments', this.env + '.json'),
+      files: [ htmlName ],
+      patterns: this._configReplacePatterns()
+    });
+  }
+
+  _cachedConfigTree: any;
+
+  _configTree() {
+    if (this._cachedConfigTree) {
```
Comment (dgeb) [inline lib/broccoli/glimmer-app.ts]: private or protected

File: lib/broccoli/glimmer-app.ts [hunk 5]
```diff
@@ -0,0 +1,358 @@
...
+ * @extends EmberApp
+ * @constructor
+ * @param {Object} [defaults]
+ * @param {Object} [options={}] Configuration options
+ */
+class GlimmerApp {
+  public options: GlimmerAppOptions;
+  public project: Project;
+  public name: string;
+  public env: 'production' | 'development';
```
Comment (josemarluedke) [inline lib/broccoli/glimmer-app.ts]: Should `test` be allowed as well?

**Comment by dgeb** [comment]
Nice refactoring with some room for more improved typings / visibility declarations to be made in the future. I don't see any obvious oversights, so I'm fine merging this when you're ready @tomdale.

**Pull request merged**